libvast/src/arrow_table_slice.cpp

```diff
@@ -611,29 +611,37 @@ const record_type& arrow_table_slice<FlatBuffer>::layout() const noexcept {
 
 template <class FlatBuffer>
 table_slice::size_type arrow_table_slice<FlatBuffer>::rows() const noexcept {
-  return record_batch()->num_rows();
+  if (auto&& batch = record_batch())
+    return batch->num_rows();
+  return {};
 }
 
 template <class FlatBuffer>
 table_slice::size_type arrow_table_slice<FlatBuffer>::columns() const noexcept {
-  return record_batch()->num_columns();
+  if (auto&& batch = record_batch())
+    return batch->num_columns();
+  return {};
 }
 
 // -- data access ------------------------------------------------------------
 
 template <class FlatBuffer>
 void arrow_table_slice<FlatBuffer>::append_column_to_index(
   id offset, table_slice::size_type column, value_index& index) const {
-  auto f = index_applier{offset, index};
-  auto array = record_batch()->column(detail::narrow_cast<int>(column));
-  decode(layout().fields[column].type, *array, f);
+  if (auto&& batch = record_batch()) {
+    auto f = index_applier{offset, index};
+    auto array = batch->column(detail::narrow_cast<int>(column));
+    decode(layout().fields[column].type, *array, f);
+  }
```

Comment:
I think we should print a warning when this fails.

Reply:
I'm not quite sure. We already print an error when the decoding fails above.

Reply:
Okay, then disregard.